packages/firestore/src/api/cache_config.ts

```diff
@@ -0,0 +1,302 @@
+  settings?: IndexedDbCacheSettings
+): IndexedDbLocalCache {
+  return new IndexedDbLocalCacheImpl(settings);
+}
+
+/**
+ * A tab manager supportting only one tab, no synchronization will be
+ * performed across tabs.
+ */
+export type IndexedDbSingleTabManager = {
+  kind: 'indexedDbSingleTab';
+  /**
+   * @internal
+   */
+  _initialize: (
+    settings: Omit<IndexedDbCacheSettings, 'tabManager'> | undefined
+  ) => void;
+  /**
+   * @internal
+   */
+  _onlineComponentProvider?: OnlineComponentProvider;
+  /**
+   * @internal
+   */
+  _offlineComponentProvider?: OfflineComponentProvider;
+};
+
+class SingleTabManagerImpl implements IndexedDbSingleTabManager {
+  kind: 'indexedDbSingleTab' = 'indexedDbSingleTab';
+
+  /**
+   * @internal
+   */
+  _onlineComponentProvider?: OnlineComponentProvider;
+  /**
+   * @internal
+   */
+  _offlineComponentProvider?: OfflineComponentProvider;
+
+  constructor(private forceOwnership?: boolean) {}
+
+  toJSON(): {} {
+    return { kind: this.kind };
+  }
+
+  /**
+   * @internal
+   */
+  _initialize(
+    settings: Omit<IndexedDbCacheSettings, 'tabManager'> | undefined
+  ): void {
+    this._onlineComponentProvider = new OnlineComponentProvider();
+    this._offlineComponentProvider = new IndexedDbOfflineComponentProvider(
+      this._onlineComponentProvider,
+      settings?.cacheSizeBytes,
+      this.forceOwnership
+    );
+  }
+}
+
+/**
+ * A tab manager supportting multiple tabs. SDK will synchronize queries and
+ * mutations done across all tabs using the SDK.
+ */
+export type IndexedDbMultipleTabManager = {
+  kind: 'IndexedDbMultipleTab';
+  /**
+   * @internal
+   */
+  _initialize: (settings: Omit<IndexedDbCacheSettings, 'tabManager'>) => void;
+  /**
+   * @internal
+   */
+  _onlineComponentProvider?: OnlineComponentProvider;
+  /**
+   * @internal
+   */
+
+  _offlineComponentProvider?: OfflineComponentProvider;
+};
+
+class MultiTabManagerImpl implements IndexedDbMultipleTabManager {
+  kind: 'IndexedDbMultipleTab' = 'IndexedDbMultipleTab';
+
+  /**
+   * @internal
+   */
+  _onlineComponentProvider?: OnlineComponentProvider;
+  /**
+   * @internal
+   */
+  _offlineComponentProvider?: OfflineComponentProvider;
+
+  toJSON(): {} {
+    return { kind: this.kind };
+  }
+
+  /**
+   * @internal
+   */
+  _initialize(
+    settings: Omit<IndexedDbCacheSettings, 'tabManager'> | undefined
+  ): void {
+    this._onlineComponentProvider = new OnlineComponentProvider();
+    this._offlineComponentProvider = new MultiTabOfflineComponentProvider(
+      this._onlineComponentProvider,
+      settings?.cacheSizeBytes
+    );
+  }
+}
+
+/**
+ * A union of all avaialbe tab managers.
+ */
+export type IndexedDbTabManager =
+  | IndexedDbSingleTabManager
+  | IndexedDbMultipleTabManager;
+
+/**
+ * Type to configure an `IndexedDbSingleTabManager` instace.
+ */
+export type IndexedDbSingleTabManagerSettings = {
+  /**
+   * Whether to force enable indexeddb cache for the client. This cannot be used
```

Comment:
"force-enable IndexedDB"

Note hyphen and capitalization of IndexedDB.... Apply capitalization here and in other occurrences in this paragraph.

Reply:
Done.